catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/impl/AssociatedQueryMetacardsHandler.java

```diff
@@ -0,0 +1,93 @@
+
+import com.google.common.collect.ImmutableSet;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.filter.FilterBuilder;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.codice.ddf.catalog.ui.metacard.workspace.QueryMetacardTypeImpl;
+import org.codice.ddf.catalog.ui.metacard.workspace.WorkspaceConstants;
+import org.codice.ddf.catalog.ui.util.EndpointUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AssociatedQueryMetacardsHandler extends AbstractAssociatedMetacardsHandler {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(AssociatedQueryMetacardsHandler.class);
+
+  private static final Set<String> METACARD_ATTRIBUTES =
+      ImmutableSet.of(
+          QueryMetacardTypeImpl.QUERY_CQL,
+          QueryMetacardTypeImpl.QUERY_FEDERATION,
+          QueryMetacardTypeImpl.QUERY_SORTS,
+          QueryMetacardTypeImpl.QUERY_FILTER_TREE);
+
+  public AssociatedQueryMetacardsHandler(
+      CatalogFramework catalogFramework, EndpointUtil util, FilterBuilder filterBuilder) {
+    super(catalogFramework, util, filterBuilder);
+  }
+
+  @Override
+  void create(Metacard existing, Metacard updated, List<Metacard> queries) {
+    LOGGER.trace("Updating query metacard associations in workspace {}", existing.getId());
+    updated.setAttribute(
+        new AttributeImpl(
+            WorkspaceConstants.WORKSPACE_ASSOCIATED_QUERIES, getMetacardIds(queries)));
+
+    LOGGER.trace("Creating new query metacards in workspace {}", existing.getId());
+    List<Metacard> unassociatedMetacards =
+        queries
+            .stream()
+            .filter(
+                query ->
+                    !isAssociated(existing, query, WorkspaceConstants.WORKSPACE_ASSOCIATED_QUERIES))
+            .collect(Collectors.toList());
+
+    createMetacard(unassociatedMetacards);
+  }
+
+  @Override
+  void update(Metacard existing, Metacard updated, List<Metacard> queries) {
+    LOGGER.trace("Updating existing query metacards in workspace {}", existing.getId());
+    List<Metacard> changedMetacards =
+        queries
+            .stream()
+            .filter(
+                query ->
+                    isAssociated(existing, query, WorkspaceConstants.WORKSPACE_ASSOCIATED_QUERIES))
+            .filter(query -> hasChanges(query.getId(), query, METACARD_ATTRIBUTES))
+            .collect(Collectors.toList());
+
+    updateMetacard(changedMetacards);
+  }
+
+  @Override
+  void delete(Metacard existing, Metacard updated, List<Metacard> queries) {
+    LOGGER.trace("Removing deleted query metacards from workspace {}", existing.getId());
+    String[] deletedIds =
+        getAssociatedMetacardIds(existing, WorkspaceConstants.WORKSPACE_ASSOCIATED_QUERIES)
+            .stream()
+            .filter(id -> !getMetacardIds(queries).contains(id))
```

Comment:
Save `getMetacardIds(queries)` to a local variable before the stream so the metacard IDs don't have to be fetched in each iteration.